nemo_skills/inference/server/code_execution_model.py

```diff
@@ -86,6 +85,16 @@ def _generate_single(
 
         if stop_phrases is None:
             stop_phrases = []
+
+        max_code_executions = self.config.max_code_executions
+        if max_code_executions is None:
+            # Assuming that openmath/math-tir prompt is used
+            code_executions = re.search(r"You may perform up to (\d+) Python code calls", prompt)
```

Comment:
this logic is a bit brittle and prompt dependent. I'd rather not have it in a class that's supposed to be generic and work with any models. Maybe instead we add a flag to set max_code_executions = remaining_code_executions to inference/generate.py? Or we can do this as a postprocessing step and let it generate more code blocks